OpenSim/Actuators/Millard2012EquilibriumMuscle.cpp

```diff
@@ -425,111 +425,74 @@ computeInitialFiberEquilibrium(SimTK::State& s) const
-                "equal to the optimal fiber length.\n\n" << endl;
-        setActuation(s, 0);
-        setFiberLength(s,getOptimalFiberLength());
+    // Initialize activation as specified by the user.
+    double clampedActivation = clampActivation(getActivation(s));
+    setActivation(s,clampedActivation);
+
+    // Initialize the multibody system to the initial state vector.
+    setFiberLength(s, getOptimalFiberLength());
+    _model->getMultibodySystem().realize(s, SimTK::Stage::Velocity);
+
+    // Compute an initial muscle state that develops the desired force and
+    // shares the muscle stretch between the muscle fiber and the tendon
+    // according to their relative stiffnesses.
+    double solnErr       = SimTK::NaN;
+    int iterations       = 0;
+    double fiberLength   = SimTK::NaN;
+    //double fiberVelocity = SimTK::NaN;
+    double tendonForce   = SimTK::NaN;
+
+    // tol is the desired tolerance in Newtons.
+    double tol = 1e-8*getMaxIsometricForce();
+    if(tol < SimTK::SignificantReal*10) {
+        tol = SimTK::SignificantReal*10;
+    }
+    int maxIter = 200;
+    double pathLength = getLength(s);
+    double pathLengtheningSpeed = getLengtheningSpeed(s);
+
+    SimTK::Vector result = SimTK::Vector(5);
+    ResultOfEstimateMuscleFiberState flag_status = estimateMuscleFiberState(
+        result, clampedActivation, pathLength, pathLengtheningSpeed, tol,
+        maxIter);
+
+    solnErr         = result[0];
+    iterations      = (int)result[1];
+    fiberLength     = result[2];
+    //fiberVelocity   = result[3];
+    tendonForce     = result[4];
+
+    switch(flag_status) {
+
+    case ResultOfEstimateMuscleFiberState::Success_Converged:
+        setActuation(s, tendonForce);
+        setFiberLength(s, fiberLength);
+        break;
+
+    case ResultOfEstimateMuscleFiberState::Warning_FiberAtLowerBound:
+        printf("\n\nMillard2012EquilibriumMuscle initialization:"
+               " %s is at its minimum fiber length of %f\n",
+               getName().c_str(), fiberLength);
+        setActuation(s, tendonForce);
+        setFiberLength(s, fiberLength);
+        break;
+
+    case ResultOfEstimateMuscleFiberState::Failure_MaxIterationsReached:
+        // Report internal variables and throw exception.
+        const char* fmt = "\n\n"
+            "  Solution error %e exceeds tolerance of %e\n"
+            "  Newton iterations reached limit of %d\n"
+            "  Clamped activation is %f\n"
+            "  Fiber length is %f\n";
+        const int messageLength = std::snprintf(nullptr, 0, fmt,
+            abs(solnErr), tol, maxIter, clampedActivation, fiberLength);
+        std::vector<char> buf(messageLength+1); //+1 for null terminator
+        std::snprintf(&buf[0], buf.size(), fmt,
+            abs(solnErr), tol, maxIter, clampedActivation, fiberLength);
+        OPENSIM_THROW_FRMOBJ( MuscleCannotEquilibrate,
+                              string(buf.begin(), buf.end()) );
+        break;
     }
```

Comment:
> Would be better to replace C code with C++ code using `std::string`, `std::stringstream` and so on. It will improve readability as well as correctness.

Done. My initial change was to remove the fixed 999-character buffer, but yes ostringstream looks better.